docs/features/latency.md

```diff
@@ -0,0 +1,231 @@
+
+we have then:
+
+* `start_time = prg_delay[0] + net_delay[0] + snd_connect_time`
+
+Note important thing: `start_time` is not the time of arrival of the first packet,
+but that time taken backwards by using the delay already recorded in TS. As TS should
+represent the delay towards `snd_connect_time`, `start_time` should be simply the same
+as `snd_connect_time`, just on the receiver side, and so obviously shifted by the
+first packet's delays of `prg_delay` and `net_delay`.
+
+So, as we have the start time defined, the above formulas:
+
+* `ETS[x] = start_time + TS[x]`
+* `PTS[x] = ETS[x] + LATENCY + DRIFT`
+
+define now the packet delivery time as:
+
+* `PTS[x] = start_time + TS[x] + LATENCY + DRIFT`
+
+and after replacing the start time we have:
+
+* `PTS[x] = prg_delay[0] + net_delay[0] + snd_connect_time + TS[x] + LATENCY + DRIFT`
+
+and for the formula of TS we get STS, so we replace it:
+
+* `PTS[x] = prg_delay[0] + net_delay[0] + STS[x] + LATENCY + DRIFT`
+
+so the true network latency in SRT we can get by moving STS to the other side:
+
+* `PTS[x] - STS[x] = prg_delay[0] + net_delay[0] + LATENCY + DRIFT`
+
+
+The DRIFT
+=========
+
+The DRIFT, for simplifyint the calculations above, should be treated as 0,
+which is the initial state. In time, however, it gets changed basing on the
+value of the Arrival Time Deviation:
+
+* `ATD[x] = ATS[x] - ETS[x]`
+
+The drift is then formed as:
+
+* `DRIFT[x] = average(ATD[x-N] ... ATD[x])`
+
+The value of the drift is tracked by appropriate number of samples and if
+it exceeds a threshold value, the drift value is applied to modify the
+base time. However, as you can see from the formula for ATD, the drift is
+simply taken from the real time when the packet was arrived, and the time
+when it would arrive, if the `prg_delay` and `net_delay` values were
+exactly the same as for the very first packet. ATD then represents the
+changes in these values. There can be two main factors that could result
+in having this value as nonzero:
+
+1. There has been observed a phenomenon in several types of networks that
+the very first packet arrives very quickly, but then as the data packets
+come in regularly, the network delay slightly increases and then stays
+for a long time with this increased value. This phenomenon could be
+mitigated by having a reliable value of RTT, so once it's observed as
+increased, a special factor could be used to decrease the positive value
+of the drift, but this currently isn't implemented. This phenomenon also
+isn't observed in every network, especially in a longer distance.
+
+2. The clock speed on both machines isn't exactly the same, which means
+that if you decipher the ETS basing on the TS, after time it may result
+in values that even precede the STS (and this way suggesting as if the
+network delay was negative) or having an enormous delay (with ATS exceeding
+PTS). This is actually the main reason of tracking the drift.
```

Comment:
```suggestion
The value of the drift is tracked over and appropriate number of samples. If
it exceeds a threshold value, the drift value is applied to modify the
base time. However, as you can see from the formula for ATD, the drift is
simply taken from the actual time when the packet arrived, and the time
when it would have arrived if the `prg_delay` and `net_delay` values were
exactly the same as for the very first packet. ATD then represents the
changes in these values. There can be two main factors that could result
in having this value as non-zero:

1. A phenomenon has been observed in several types of networks where
the very first packet arrives quickly, but as subsequent data packets
come in regularly, the network delay slightly increases and then remains fixed
for a long time at this increased value. This phenomenon can be
mitigated by having a reliable value for RTT. Once the increase is observed 
a special factor could be applied to decrease the positive value
of the drift. This isn't currently implemented. This phenomenon also
isn't observed in every network, especially those covering longer distances.

2. The clock speed on both machines (sender and receiver) isn't exactly the same, 
which means that if you decipher the ETS basing on the TS, over time it may result
in values that even precede the STS (suggesting a negative network delay) or that 
have an enormous delay (with ATS exceeding PTS). This is actually the main reason 
for tracking the drift.
```